memphis/memphis.go

```diff
@@ -103,21 +103,27 @@ func CreateFunction(eventHandler HandlerType, options ...PayloadOption) {
 
 			var handlerInput any
 			if params.UserStruct != nil {
-				UnmarshalIntoStruct(payload, params.UserStruct)
+				err = UnmarshalIntoStruct(payload, params.UserStruct)
 				handlerInput = params.UserStruct
 			} else {
 				handlerInput = payload
 			}
-
-			modifiedPayload, modifiedHeaders, err := params.Handler(handlerInput, msg.Headers, event.Inputs)
-			_, ok := modifiedPayload.([]byte)
+			
+			var modifiedPayload any
+			var modifiedHeaders map[string]string
+			var ok bool
+
+			if err == nil{
+				modifiedPayload, modifiedHeaders, err = params.Handler(handlerInput, msg.Headers, event.Inputs)
+				_, ok = modifiedPayload.([]byte)
+			}
 
 			if err == nil && !ok {
```

Comment:
move the check if !ok to be after `_, ok = modifiedPayload.([]byte)` - inside the first `if err == nil{`

Reply:
Moved this with that last commit [83a8d99](https://github.com/memphisdev/memphis-functions.go/pull/8/commits/83a8d99a449810c7f9fde0e28aaf9518dac8f20d)